### PR TITLE
Retry aws creds

### PR DESCRIPTION
**Pull request by ahilsend**

### CODE REVIEW

File: pkg/keystore/awsKeystore.go [hunk 1]
```diff
@@ -40,8 +46,26 @@ func createAwsSession(config *AwsConfig) *session.Session {
...
+func waitUntilValidSession(config *AwsConfig) *session.Session {
+	awsSession := createAwsSession(config)
+	if !config.RetryOnCredentialsError {
+		return awsSession
+	}
+
+	for {
+		_, err := awsSession.Config.Credentials.Get()
+		if err != credentials.ErrNoValidProvidersFoundInChain {
+			return awsSession
```
Comment (zegl) [inline pkg/keystore/awsKeystore.go]: Would it be better to return the error here? So that it's possible to debug why it failed.

File: main.go [hunk 2]
```diff
@@ -35,8 +35,16 @@ func getEnv(name string) string {
 }
 
 func createAwsConfig() *keystore.AwsConfig {
+	retryOnCredentialsError, _ := strconv.ParseBool(os.Getenv("AWS_RETRY_ON_CREDENTIALS_ERROR"))
```
Comment (zegl) [inline main.go]: Is there any time you wouldn't want to retry?

File: pkg/keystore/awsKeystore.go (hunk 1 above)
Comment (ahilsend) [reply pkg/keystore/awsKeystore.go]: `ErrNoValidProvidersFoundInChain` is the one I'm expecting if it fails. maybe better to check 
```go
if err == nil {
```
?

File: main.go (hunk 2 above)
Comment (ahilsend) [reply main.go]: Figured as I don't know how to test this, it's better to disable it by default for now :man_shrugging:

**Review by zegl — APPROVED** [review]

**Pull request merged**